File: interview/sample/problems/two_sum.py
def two_sum(nums, target):
    num_map = {}  # To store number and its index
    for i, num in enumerate(nums):
        complement = target - num
        if complement in num_map:
            return [num_map[complement], i]
        num_map[num] = i
      
print(two_sum([2, 7, 11, 15], 9))      # Output: [0, 1]
print(two_sum([3, 2, 4], 6))           # Output: [1, 2]
print(two_sum([3, 3], 6))              # Output: [0, 1]

File: interview/sample/problems/TwoSum.java
import java.util.HashMap;
import java.util.Map;

public class TwoSum {
    public static int[] twoSum(int[] nums, int target) {
        Map<Integer, Integer> map = new HashMap<>();
        for (int i = 0; i < nums.length; i++) {
            int complement = target - nums[i];
            if (map.containsKey(complement)) {
                return new int[] { map.get(complement), i };
            }
            map.put(nums[i], i);
        }
        throw new IllegalArgumentException("No two sum solution");
    }

    public static void main(String[] args) {
        int[] nums = {2, 7, 11, 15};
        int target = 9;
        int[] result = twoSum(nums, target);
        System.out.println("Indices: [" + result[0] + ", " + result[1] + "]");
    }
}
